murmur.go

```diff
@@ -0,0 +1,295 @@
+		fallthrough
+	case 11:
+		k2 ^= uint64(data[10]) << 16
+		fallthrough
+	case 10:
+		k2 ^= uint64(data[9]) << 8
+		fallthrough
+	case 9:
+		k2 ^= uint64(data[8]) << 0
+
+		k2 *= c2_128
+		k2 = bits.RotateLeft64(k2, 33)
+		k2 *= c1_128
+		h2 ^= k2
+
+		fallthrough
+
+	case 8:
+		k1 ^= uint64(data[7]) << 56
+		fallthrough
+	case 7:
+		k1 ^= uint64(data[6]) << 48
+		fallthrough
+	case 6:
+		k1 ^= uint64(data[5]) << 40
+		fallthrough
+	case 5:
+		k1 ^= uint64(data[4]) << 32
+		fallthrough
+	case 4:
+		k1 ^= uint64(data[3]) << 24
+		fallthrough
+	case 3:
+		k1 ^= uint64(data[2]) << 16
+		fallthrough
+	case 2:
+		k1 ^= uint64(data[1]) << 8
+		fallthrough
+	case 1:
+		k1 ^= uint64(data[0]) << 0
+		k1 *= c1_128
+		k1 = bits.RotateLeft64(k1, 31)
+		k1 *= c2_128
+		h1 ^= k1
+	}
+
+	h1 ^= uint64(clen)
+	h2 ^= uint64(clen)
+
+	h1 += h2
+	h2 += h1
+
+	h1 = fmix64(h1)
+	h2 = fmix64(h2)
+
+	h1 += h2
+	h2 += h1
+	// End of original twmb/murmur3.Sum128
+	// Save this hash, as it's equivalent of h1, h2 = hash.Write(data).Sum128()
+	res[0], res[1] = h1, h2
+
+	// Now, reset state back to before the tail calculation.
+	// The state is such that all but up to final 15 bytes have been processed, as murmur3 works
+	// on 128 bit / 16 byte chunks for all but the final mixing.
+	// last byte of the array, and apply the Sum128() logic accordingly.
+	k1, k2 = 0, 0
+	h1, h2 = _h1, _h2 // restore state to before final mixing
+	clen++            // total hash data length now includes entropy
+	// Fist, process entropy byte *only*. No fallthrough.
+	// However, if we are really lucky and remainder was 15 bytes, with entropy byte it is now a full block.
+	// In that case, just apply bmix() again, with the modified payload and directly go to hash finalization.
+	rlen := len(data)
+
+	// Check if we're lucky, and have to just perform bmix() of the final chunk
+	if rlen == 15 {
+		k1 = uint64(data[0]) | uint64(data[1])<<8 | uint64(data[2])<<16 | uint64(data[3])<<24 | uint64(data[4])<<32 | uint64(data[5])<<40 | uint64(data[6])<<48 | uint64(data[7])<<56
+		// Note: this is different from original bmix() - we don't mutate `data`,
+		// so we reuse the entire 15 byte slice of the original remainder and treat the last byte as equal to `entropy`
+		k2 = uint64(data[8]) | uint64(data[9])<<8 | uint64(data[10])<<16 | uint64(data[11])<<24 | uint64(data[12])<<32 | uint64(data[13])<<40 | uint64(data[14])<<48 | entropy<<56
+
+		k1 *= c1_128
+		k1 = bits.RotateLeft64(k1, 31)
+		k1 *= c2_128
+		h1 ^= k1
+
+		h1 = bits.RotateLeft64(h1, 27)
+		h1 += h2
+		h1 = h1*5 + 0x52dce729
+
+		k2 *= c2_128
+		k2 = bits.RotateLeft64(k2, 33)
+		k2 *= c1_128
+		h2 ^= k2
+
+		h2 = bits.RotateLeft64(h2, 31)
+		h2 += h1
+		h2 = h2*5 + 0x38495ab5
+		// No further processing needed, skip the next state
+		goto Finalize
+	}
+
+	// Process entropy byte - note, **no fallthrough**, because
+	// how we process it depends on how long the payload is
+	switch rlen + 1 {
+	case 15:
+		k2 ^= entropy << 48
+		goto l14
```

Comment:
Nit: maybe a comment to the effect of:

// the entropy byte is the last byte (index 14)--process it, then proceed with the rest of the array (`length 14` case below).